addons/point_of_sale/models/pos_order.py

```diff
@@ -143,7 +145,7 @@ def _process_order(self, pos_order):
                 cash_journal_id = cash_journal[0].id
             order.add_payment({
                 'amount': -pos_order['amount_return'],
-                'payment_date': fields.Datetime.now(),
+                'payment_date': payment_date,
```

Comment:
Is more simply, only  fields.Date.context_today(self) instead fields.Datetime.now().